### PR TITLE
specextract logic clean up

### PR DESCRIPTION
**Pull request by nplee**
This update includes the logic clean up needed to eliminate many of the IDLisms in specextract.  This is an intermediary point in the script rewrite where some new functionality is disabled and/or commented out in the script.  Off-ramping at this point shows no differences in regression testing with prior versions of the script and will immediately address #400 although error message formatting has been changed.

### CODE REVIEW

File: bin/specextract [hunk 1]
```diff
@@ -609,9 +463,9 @@ def set_badpix(evtfile, bpixfile, instrument, verbose):
 
         finally:
             if os.path.isfile(bpixfile) and getattr(ardlib,bpixpar.replace("-","_")).rstrip("[BADPIX]") == bpixfile:
```
Comment (DougBurke) [inline bin/specextract:465]: I think this is better written as

    if not os.path.isfile(bpixfile) or getattr(...) != bpixfile:
        raise IOError(...)

as it is a bit clearer what leads to an error.

File: bin/specextract [hunk 2]
```diff
@@ -798,28 +652,113 @@ def group_by_obsid(file_stack, file_type):
...
+              be explicitly purged and likewise if they are covered by a 
+              large flux, then they will be preserved and the ARF 
+              appropriately weighted.
+    """
+
+    cr = pcr.read_file("{}[wmap]".format(wmapfile))
+    im = cr.get_image().values
+    threshold = float(threshold)
+
+    cr.__del__()
```
Comment (DougBurke) [inline bin/specextract:706]: I'm not sure what deleting the crate does to the values - I would suggest either

    im = cr.get_image().values.copy()

or just remove the `cr.__del__()` call (I prefer the second option).

File: bin/specextract [hunk 3]
```diff
@@ -798,28 +652,113 @@ def group_by_obsid(file_stack, file_type):
...
+    dmimgthresh.cut = cutoff
+    dmimgthresh.value = "0.0"
+    dmimgthresh.expfile = ""
+    dmimgthresh.clobber = "yes"
+    dmimgthresh.verbose = "0"
+
+    dmimgthresh()
+
+    threshfile.close()
+    del(im)
```
Comment (DougBurke) [inline bin/specextract:757]: If you really want, add the `del(cr)` here ;-)

File: bin/specextract [hunk 4]
```diff
@@ -1122,10 +1057,10 @@ def event_stats(file, colname):
 
     elif str(colname) in ["x","y","chipx","chipy"]:
```
Comment (DougBurke) [inline bin/specextract:1058]: The `str()` isn't helping us here so just say

    if colname in [...]

File: bin/specextract [hunk 5]
```diff
@@ -1135,18 +1070,21 @@ def event_stats(file, colname):
...
 
-        elif colname=="chipx":
+        elif colname == "chipx":
             return int(float(src_x))
 
-        elif colname=="chipy":
+        elif colname == "chipy":
             return int(float(src_y))
 
+    cr.__del__()
```
Comment (DougBurke) [inline bin/specextract:1085]: We should not need this - there was some code way back in CIAO 4.<very-small-number> which did, but if you have evidence we still need this then we need a cratedev bug report!

**Comment by DougBurke** [comment]
@nplee - feel free when writing strings to use the f-string support added in Python 3.6 - ie change

    "foo={}  bar={}".format(foo, bar)

to

    f"foo={foo} bar={bar}"

It's not worth changing code that already exists, but as you come across new code (or edit old code) don't forget this trick!

**Comment by nplee** [comment]
@DougBurke updated as we had discussed with your comments.

**Comment by DougBurke** [comment]
Cheers

**Pull request merged**